ssl/handshake_server.cc

```diff
@@ -291,13 +291,19 @@ bool ssl_parse_client_cipher_list(
       return false;
     }
 
+    // Storing only supported ciphers
     const SSL_CIPHER *c = SSL_get_cipher_by_value(cipher_suite);
     if (c != NULL && !sk_SSL_CIPHER_push(sk.get(), c)) {
       OPENSSL_PUT_ERROR(SSL, SSL_R_ERROR_IN_RECEIVED_CIPHER_LIST);
       return false;
     }
   }
 
+  // Store raw bytes for cipher suites offered
+  ssl->all_client_cipher_suites.reset(static_cast<char *>(OPENSSL_memdup(
```

Comment:
Should this be stored earlier in the function (prior to parsing) in case parsing fails?  Is there a use-case where a user would want to inspect/log the client hello cipher suites message to diagnose a failure?